src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeGenerator.cs

```diff
@@ -265,7 +265,7 @@ protected override void OnExpandPopulateFolders(IList<TreeNode> currentChildren,
                 NodeType = "Folder",
                 NodeTypeId = NodeTypes.ServerLevelServerRoles,
                 IsSystemObject = false,
-                ValidFor = ValidForFlag.AllOnPrem|ValidForFlag.SqlOnDemand,
+                ValidFor = ValidForFlag.AllOnPrem,
```

Comment:
Is the SqlOnDemand flag being used elsewhere? If not remove it

Reply:
Yup, it is used on several places in the same file.